bot/cogs/utils.py

```diff
@@ -51,71 +53,37 @@ def __init__(self, bot: Bot):
 
         self.base_pep_url = "http://www.python.org/dev/peps/pep-"
         self.base_github_pep_url = "https://raw.githubusercontent.com/python/peps/master/pep-"
+        self.peps_listing_api_url = "https://api.github.com/repos/python/peps/contents?ref=master"
 
-    @command(name='pep', aliases=('get_pep', 'p'))
-    async def pep_command(self, ctx: Context, pep_number: str) -> None:
-        """Fetches information about a PEP and sends it to the channel."""
-        if pep_number.isdigit():
-            pep_number = int(pep_number)
-        else:
-            await ctx.invoke(self.bot.get_command("help"), "pep")
-            return
-
-        # Handle PEP 0 directly because it's not in .rst or .txt so it can't be accessed like other PEPs.
-        if pep_number == 0:
-            return await self.send_pep_zero(ctx)
-
-        possible_extensions = ['.txt', '.rst']
-        found_pep = False
-        for extension in possible_extensions:
-            # Attempt to fetch the PEP
-            pep_url = f"{self.base_github_pep_url}{pep_number:04}{extension}"
-            log.trace(f"Requesting PEP {pep_number} with {pep_url}")
-            response = await self.bot.http_session.get(pep_url)
+        self.peps: Dict[int, str] = {}
+        self.refresh_peps_urls.start()
 
-            if response.status == 200:
-                log.trace("PEP found")
-                found_pep = True
+    @loop(hours=3)
+    async def refresh_peps_urls(self) -> None:
+        """Refresh PEP URLs listing in every 3 hours."""
+        # Wait until HTTP client is available
+        await self.bot.wait_until_guild_available()
 
-                pep_content = await response.text()
+        async with self.bot.http_session.get(self.peps_listing_api_url) as resp:
+            listing = await resp.json()
 
-                # Taken from https://github.com/python/peps/blob/master/pep0/pep.py#L179
-                pep_header = HeaderParser().parse(StringIO(pep_content))
+        for file in listing:
+            name = file["name"]
+            if name.startswith("pep-") and (name.endswith(".txt") or name.endswith(".rst")):
```

Comment:
endswith can accept a tuple of strings to check so the or can be removed and a single endswith used